db/schema.sql

```diff
@@ -0,0 +1,609 @@
+-- schema definition
+
+#########################
+#
+# DDL Section
+#
+#########################
+
+#  OfferingType
+
+DROP TABLE IF EXISTS OfferingType;
+
+CREATE TABLE IF NOT EXISTS OfferingType (
+  id          INT PRIMARY KEY AUTO_INCREMENT NOT NULL,
+  name        VARCHAR(255) COMMENT 'starbucks refreshers beverages',
+  description VARCHAR(255) COMMENT 'longer days mean more time to sip something warm'
+
+);
+
+# Offering
+
+DROP TABLE IF EXISTS Offering;
+
+CREATE TABLE Offering (
+  id             INT PRIMARY KEY NOT NULL AUTO_INCREMENT,
+  name           VARCHAR(255) COMMENT 'cool lime starbucks refreshers beverage',
+  description    VARCHAR(255),
+  offeringTypeId INT             NOT NULL
+);
+
+ALTER TABLE Offering
+  ADD FOREIGN KEY (offeringTypeId) REFERENCES OfferingType (id);
+
+# Tag
+
+DROP TABLE IF EXISTS Tag;
+
+CREATE TABLE IF NOT EXISTS Tag (
+  id   INT PRIMARY KEY AUTO_INCREMENT NOT NULL,
+  name VARCHAR(255) COMMENT 'name can be based on tags'
+
+);
+
+# Location
+
+DROP TABLE IF EXISTS Location;
+
+CREATE TABLE IF NOT EXISTS Location (
+  id      INT PRIMARY KEY AUTO_INCREMENT NOT NULL,
+  name    VARCHAR(255) COMMENT 'name of places',
+  address VARCHAR(1024),
+  phone   INT
+
+);
+
+# Ingredient
+
+DROP TABLE IF EXISTS Ingredient;
+
+CREATE TABLE IF NOT EXISTS Ingredient (
+  id   INT PRIMARY KEY AUTO_INCREMENT NOT NULL,
+  name VARCHAR(255) COMMENT 'lime refresher base'
+);
+
+DROP TABLE IF EXISTS OfferingTag;
+
+# Offering <--> Tag
+
+CREATE TABLE IF NOT EXISTS OfferingTag (
+  offeringId INT NOT NULL,
+  tagId      INT NOT NULL
+);
+
+ALTER TABLE OfferingTag
+  ADD FOREIGN KEY (offeringId) REFERENCES Offering (id);
+ALTER TABLE OfferingTag
+  ADD FOREIGN KEY (tagId) REFERENCES Tag (id);
```

Comment:
Can we change the foreign key from tagid to tagname? This would simplify our query, since we are getting an array of tag strings from the UI.
`With offeringtagid and tagid as mappings`
``` 
select 
    t.name
from
    offeringtag ot,
    tag t
where
    t.name like '%lime%'
        and t.name like '%fruit%'
	t.id = ot.id;
```
vs
`With tagname as foreign key from offeringtag, referencing tag table`
```
select 
    name
from
    offeringtag
where
    tagname like '%lime%'
        or tagname like '%fruit%';
```

Reply:
While this would make it easier to query, it would be incorrect, both technically and semantically.

Reply:
Also, the whole point of substituting something like `"fruit"` with something like `34` is to reduce data redundancy (5 bytes in the former case, and at most 4 bytes in the latter).